### ISSUE TITLE
java.lang.VerifyError: Verifier rejected class it.fast4x.rimusic.ui.screens.player.PlayerModernKt

### ISSUE THREAD

**Issue by twistios** [fast4x/RiMusic#2913]
### Steps to reproduce the bug

1. Start the app
2. Play a song

### Expected behavior

The song plays.

### Actual behavior

The app crashes with java.lang.VerifyError.

### Logs

<details><summary>Details</summary>
<p>

````
Process: it.fast4x.rimusic.debug, PID: 29728
                                                                                                    java.lang.VerifyError: Verifier rejected class it.fast4x.rimusic.ui.screens.player.PlayerModernKt: void it.fast4x.rimusic.ui.screens.player.PlayerModernKt.PlayerModern(androidx.navigation.NavController, it.fast4x.rimusic.ui.screens.player.PlayerSheetState, androidx.compose.material3.SheetState, androidx.compose.ui.Modifier, androidx.compose.foundation.shape.RoundedCornerShape, kotlin.jvm.functions.Function0, androidx.compose.runtime.Composer, int, int) failed to verify: void it.fast4x.rimusic.ui.screens.player.PlayerModernKt.PlayerModern(androidx.navigation.NavController, it.fast4x.rimusic.ui.screens.player.PlayerSheetState, androidx.compose.material3.SheetState, androidx.compose.ui.Modifier, androidx.compose.foundation.shape.RoundedCornerShape, kotlin.jvm.functions.Function0, androidx.compose.runtime.Composer, int, int): [0x1F8B] copy1 v233<-v48 type=Imprecise Constant: 127 cat=3 (declaration of 'it.fast4x.rimusic.ui.screens.player.PlayerModernKt' appears in /data/data/it.fast4x.rimusic.debug/code_cache/.overlay/base.apk/classes18.dex)
                                                                                                    	at it.fast4x.rimusic.MainActivity$startApp$5$4$1$4.invoke(MainActivity.kt:925)
                                                                                                    	at it.fast4x.rimusic.MainActivity$startApp$5$4$1$4.invoke(MainActivity.kt:924)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:118)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at it.fast4x.rimusic.ui.components.CustomModalBottomSheetKt$CustomModalBottomSheet$2.invoke(CustomModalBottomSheet.kt:61)
                                                                                                    	at it.fast4x.rimusic.ui.components.CustomModalBottomSheetKt$CustomModalBottomSheet$2.invoke(CustomModalBottomSheet.kt:59)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:118)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.material3.ModalBottomSheetKt$ModalBottomSheetContent$7.invoke(ModalBottomSheet.kt:336)
                                                                                                    	at androidx.compose.material3.ModalBottomSheetKt$ModalBottomSheetContent$7.invoke(ModalBottomSheet.kt:284)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.material3.SurfaceKt$Surface$1.invoke(Surface.kt:126)
                                                                                                    	at androidx.compose.material3.SurfaceKt$Surface$1.invoke(Surface.kt:108)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:380)
                                                                                                    	at androidx.compose.material3.SurfaceKt.Surface-T9BRK9s(Surface.kt:105)
                                                                                                    	at androidx.compose.material3.ModalBottomSheetKt.ModalBottomSheetContent-IQkwcL4(ModalBottomSheet.kt:216)
                                                                                                    	at androidx.compose.material3.ModalBottomSheetKt$ModalBottomSheet$3.invoke(ModalBottomSheet.kt:173)
                                                                                                    	at androidx.compose.material3.ModalBottomSheetKt$ModalBottomSheet$3.invoke(ModalBottomSheet.kt:166)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.material3.ModalBottomSheet_androidKt$ModalBottomSheetDialog$dialog$1$1$1.invoke(ModalBottomSheet.android.kt:300)
                                                                                                    	at androidx.compose.material3.ModalBottomSheet_androidKt$ModalBottomSheetDialog$dialog$1$1$1.invoke(ModalBottomSheet.android.kt:296)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.material3.ModalBottomSheetDialogLayout.Content(ModalBottomSheet.android.kt:354)
                                                                                                    	at androidx.compose.ui.platform.AbstractComposeView$ensureCompositionCreated$1.invoke(ComposeView.android.kt:259)
2024-07-25 00:10:22.716 29728-29728 AndroidRuntime          it.fast4x.rimusic.debug              E  	at androidx.compose.ui.platform.AbstractComposeView$ensureCompositionCreated$1.invoke(ComposeView.android.kt:258)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:380)
                                                                                                    	at androidx.compose.ui.platform.CompositionLocalsKt.ProvideCommonCompositionLocals(CompositionLocals.kt:216)
                                                                                                    	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt$ProvideAndroidCompositionLocals$3.invoke(AndroidCompositionLocals.android.kt:132)
                                                                                                    	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt$ProvideAndroidCompositionLocals$3.invoke(AndroidCompositionLocals.android.kt:131)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:380)
                                                                                                    	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt.ProvideAndroidCompositionLocals(AndroidCompositionLocals.android.kt:121)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$3.invoke(Wrapper.android.kt:155)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1$1$3.invoke(Wrapper.android.kt:154)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:401)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1$1.invoke(Wrapper.android.kt:154)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1$1.invoke(Wrapper.android.kt:133)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.ActualJvm_jvmKt.invokeComposable(ActualJvm.jvm.kt:97)
                                                                                                    	at androidx.compose.runtime.ComposerImpl.doCompose(Composer.kt:3593)
                                                                                                    	at androidx.compose.runtime.ComposerImpl.composeContent$runtime_release(Composer.kt:3520)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.composeContent(Composition.kt:743)
                                                                                                    	at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(Recomposer.kt:1114)
                                                                                                    	at androidx.compose.runtime.ComposerImpl$CompositionContextImpl.composeInitial$runtime_release(Composer.kt:3874)
                                                                                                    	at androidx.compose.runtime.ComposerImpl$CompositionContextImpl.composeInitial$runtime_release(Composer.kt:3874)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.composeInitial(Composition.kt:649)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:635)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:133)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:124)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.setOnViewTreeOwnersAvailable(AndroidComposeView.android.kt:1597)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition.setContent(Wrapper.android.kt:124)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition.onStateChanged(Wrapper.android.kt:180)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.jvm.kt:320)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.jvm.kt:198)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:131)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:124)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.onAttachedToWindow(AndroidComposeView.android.kt:1678)
                                                                                                    	at android.view.View.dispatchAttachedToWindow(View.java:23227)
                                                                                                    	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3698)
2024-07-25 00:10:22.718 29728-29728 AndroidRuntime          it.fast4x.rimusic.debug              E  	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3705)
                                                                                                    	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3705)
                                                                                                    	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3705)
                                                                                                    	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3705)
                                                                                                    	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3901)
                                                                                                    	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:3288)
                                                                                                    	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:11344)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1689)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1698)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1153)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1079)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1646)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230)
                                                                                                    	at android.os.Looper.loop(Looper.java:319)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8919)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
````

</p>
</details> 

### RiMusic version

master branch on commit 9ae01df20d88ef0d7fc7

### Android version

14

### Additional information

I was able to track down the bad (i.e. broken) commit: c9de5a4333726573f27ec4d195bd1148c529712a.
The previous commit a24bbd3fb5b0c86080a43b20bd1300c8d2783607 is still good.

**Comment by twistios** [comment]
I will try to debug this and do a pull request if I find the problem.

**Comment by twistios** [comment]
It seems like it does not want to have the box around the other stuff (because the issue is still there if I remove every change inside of the box). The issue is about the box-part of the commit.

**Comment by twistios** [comment]
I got a solution for the original problem, but now I get `ArrayIndexOutOfBoundsException`.

**Comment by twistios** [comment]
The original problem seems to be that `binder.player.mediaItemCount` does not exist/is not correctly formated/..., I found this by using a number instead of the player-thing.

**Comment by twistios** [comment]
Ok, this issue is very confusing. Because now the original error only occures with the code after the fling-definition...

**Comment by twistios** [comment]
Next observation: If `bottomgradient` condition is changed to `false` (i.e. the branch with  `isLandscape` can not be accessed anymore), the `VerifyError` goes away.

**Comment by twistios** [comment]
The problem with `isLandscape` can be fixed by this construction:
````
Modifier.run {
                        val v = isLandscape
                        ...
}
````
If I now put the `colorPaletteMode`-condition back in, it fails again. This can not be fixed via the run-trick.

**Comment by twistios** [comment]
For some reason, there now is a new VerifyError. It is the same type of problem (the error is the same except for different registers). This time I was not able to figure out how to get around the issue. I was able to identify one of the lines that when enabled trigger it and if disabled doesn't (this includes that some other lines are commented and that I had to change to code to isolate this problematic section).
